src/odemis/acq/stream/_sync.py

```diff
@@ -2000,3 +2001,212 @@ def _runAcquisition(self, future):
+
+        # The dwell time from the Nikon C2 will set based on what the device is capable of
+        # As a result, we need to recalculate our total dwell time based around this value
+        # and the number of frames we can compute
+        px_dt = min(self._dwellTime.value, self._scanner.dwellTime.range[1])
+        self._nfr = int(math.ceil(self._dwellTime.value / px_dt))  # number of frames
+        px_dt = self._dwellTime.value / self._nfr  # the new dwell time per frame
+        self._scanner.dwellTime.value = px_dt  # try set the C2 dwell time value again.
+        logging.info("Total dwell time: %f s, Pixel Dwell time: %f, collecting %d frames...",
+                     self._dwellTime.value, px_dt, self._nfr)
+
+        self._tc_scanner.dwellTime.value = self._dwellTime.value
+
+        return px_dt
+
+    def _setLightEmission(self, value):
+        # set all light emissions at once to a value
+        em = self._light.emissions.value
+        em = [value] * len(em)
+        self._light.emissions.value = em
+
+    def _runAcquisition(self, future=None):
+
+        logging.debug("Starting job: acquisition")
+
+        try:
+            self._new_acquisition = True
+            self._acq_done.clear()
+
+            px_dt = self._prepareHardware()
+            frame_time = px_dt * numpy.prod(self._scanner.resolution.value)
+            logging.info("Frame time: %f s", frame_time)
+
+            # Start Symphotime acquisition
+            self._time_correlator.data.subscribe(self._onAcqStop)
+
+            # Turn on the lights
+            self._setLightEmission(1)
+
+            # Start the acquisition
+            self._pdetector.data.subscribe(self._onNewData)
+            self._new_acquisition = True
+
+            # For each frame
+            for i in range(self._nfr):
+                # premature cancellation occurred if the flag has already been set.
+                if self._acq_done.is_set():
+                    raise RuntimeError("Acquisition canceled!!!")
+
+                self._frame_done.clear()
+                # turn on the light and pulse the laser (period already set)
+                logging.info("Getting frame %d", i + 1)
+
+                # wait for the measurement to run for the total dwell time
+                # this will stop blocking if the acquisition is cancelled
+                tstart = time.time()
+                if not self._frame_done.wait(3 * frame_time + 1):
+                    raise IOError("Timeout waiting for frame.")
+
+                logging.debug("waited %f s", time.time() - tstart)
+
+            # End Symphotime acq
+            # The measurement is stopped.
+            self._pdetector.data.unsubscribe(self._onNewData)
+            self._time_correlator.data.unsubscribe(self._onAcqStop)
+
+            # turn off the light
+            self._setLightEmission(0)
+
+            # Finish
+            self._acq_done.set()
+
+        except Exception as e:
+            logging.exception(e)
```

Comment:
You need to also stop the acquisition in the case of an error. Otherwise the hardware might end up in some limbo state, where some detectors are still acquiring, and some stopped due to the failure.